staging/src/k8s.io/client-go/tools/clientcmd/api/helpers.go

```diff
@@ -98,6 +98,9 @@ func ShortenConfig(config *Config) {
 		if len(authInfo.ClientCertificateData) > 0 {
 			authInfo.ClientCertificateData = redactedBytes
 		}
+		if len(authInfo.Token) > 0 {
+			authInfo.Token = string(redactedBytes)
```

Comment:
I'm surprised this works after being cast to a string.

Reply:
Yeah, It is different than the others which are byte arrays.  Token is a string.

Reply:
what does the output of this look like? we also have other potentially sensitive fields in the config (password, auth config); see corresponding sanitization of rest.Config printing in https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/rest/config.go#L158-L175

Reply:
The goal of ShortenConfig was not to keep confidential data from the output (if you're running this command, you already have read access to the data), but to prevent massively long output from byte-serialized certificates.

Reply:
Agreed that it doesn't provide security.

I think you questioning is right though.  There is a problem.  The unit test passes, but the output is not "REDACTED" it is garbage.  Good catch.  I will re-visit this.

Reply:
Sorry @lavalamp, I think your intuition was right, it doesn't work. Not sure what I was thinking... :confused: 

@lavalamp, @liggitt I pushed a new commit and the output looks like this:

```
- name: clusterUser_k8s_akstest
  user:
    client-certificate-data: REDACTED
    client-key-data: REDACTED
    token: REDACTED
```

Reply:
Why was the test wrong?

Reply:
> Why was the test wrong?

It was a new test case I added that was wrong, so it wasn't a problem in an existing test.  I made an assumption that `string(redactedBytes)` would result in "REDACTED" but that was wrong.

redactedBytes is defined as:

```
sDec, _ := base64.StdEncoding.DecodeString("REDACTED+")
redactedBytes = []byte(string(sDec))
```

I didn't look closely enough at how that actually worked.  

It is sort of tricky in that it is starting with a "base-64 encoded" value of "REDACTED+" and then decoding it, storing it in bytes, so that later when it is re-encoded, it displays as a human-readable "REDACTED".

When I did `string(redactedBytes)` I got the base64 decode of "REDACTED+" which is actually "D@�\t1\x03".

Reply:
There is a comment basically saying how the"trick" works, somehow I glossed right over it.
https://github.com/kubernetes/kubernetes/blob/fd0eb5dff627238fdb3a68e93abefc8a267a863a/staging/src/k8s.io/client-go/tools/clientcmd/api/helpers.go#L91